internal/policies/apparmor/apparmor_test.go

```diff
@@ -28,29 +31,33 @@ func TestApplyPolicy(t *testing.T) {
 		entries []entry.Entry
 		user    bool
 
-		noParserOutput          bool
+		noParserOutput         bool
+		destAlreadyExists      string
+		readOnlyApparmorDir    string
+		noApparmorParser       bool
+		noLoadedPoliciesFile   bool
+		existingLoadedPolicies []string
+
 		saveAssetsError         bool
 		removeUnusedAssetsError bool
-		apparmorParserError     bool
-		destAlreadyExists       string
-		readOnlyApparmorDir     string
-		noApparmorParser        bool
+		apparmorParserError     string
 
 		wantErr bool
 	}{
 		// computer cases
-		"computer, one profile":              {entries: defaultMachineProfile},
-		"computer, multiple profiles,":       {entries: []entry.Entry{{Key: "apparmor-machine", Value: "usr.bin.foo\nusr.bin.bar\nnested/usr.bin.baz"}}},
-		"computer, duplicated profiles":      {entries: []entry.Entry{{Key: "apparmor-machine", Value: "usr.bin.foo\nusr.bin.foo"}}},
-		"computer, blank line profiles":      {entries: []entry.Entry{{Key: "apparmor-machine", Value: "usr.bin.foo\n\nusr.bin.bar\n"}}},
-		"computer, profiles with whitespace": {entries: []entry.Entry{{Key: "apparmor-machine", Value: " usr.bin.foo\n\n usr.bin.bar   \nnested/usr.bin.baz "}}},
-		"computer, whitespace-only value":    {entries: []entry.Entry{{Key: "apparmor-machine", Value: "       "}}, noParserOutput: true},
-		"computer, only blank profiles":      {entries: []entry.Entry{{Key: "apparmor-machine", Value: "\n\n\n"}}, noParserOutput: true},
-		"existing .old directory is removed": {entries: defaultMachineProfile, destAlreadyExists: "machine.old", noParserOutput: true},
-		"existing .new directory is removed": {entries: defaultMachineProfile, destAlreadyExists: "machine.new", noParserOutput: true},
+		"computer, one profile":                    {entries: defaultMachineProfile},
+		"computer, multiple profiles,":             {entries: []entry.Entry{{Key: "apparmor-machine", Value: "usr.bin.foo\nusr.bin.bar\nnested/usr.bin.baz"}}},
+		"computer, duplicated profiles":            {entries: []entry.Entry{{Key: "apparmor-machine", Value: "usr.bin.foo\nusr.bin.foo"}}},
+		"computer, blank line profiles":            {entries: []entry.Entry{{Key: "apparmor-machine", Value: "usr.bin.foo\n\nusr.bin.bar\n"}}},
+		"computer, profiles with whitespace":       {entries: []entry.Entry{{Key: "apparmor-machine", Value: " usr.bin.foo\n\n usr.bin.bar   \nnested/usr.bin.baz "}}},
+		"computer, whitespace-only value":          {entries: []entry.Entry{{Key: "apparmor-machine", Value: "       "}}, noParserOutput: true},
+		"computer, only blank profiles":            {entries: []entry.Entry{{Key: "apparmor-machine", Value: "\n\n\n"}}, noParserOutput: true},
+		"computer, previous profiles are unloaded": {destAlreadyExists: "machine", existingLoadedPolicies: []string{"/usr/bin/foo", "/usr/bin/bar", "/usr/bin/baz"}, entries: defaultMachineProfile},
```

Comment:
The entries here are listed in the end. I think it's better to keep the input pattern that you were using above (won't matter if you agree with the other suggestion).